Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -0,0 +1,457 @@
+    """ Dedup events response.
+    Armis API V.1.8 supports time filtering in requests only up to level of seconds (and not milliseconds).
+    Therefore, if there are more events with the same timestamp than in the current fetch cycle,
+    additional handling is necessary.
+
+    Cases:
+    1.  Empty event list (no new events received from API response).
+        Meaning: Usually means there are not any more events to fetch at the moment.
+        Handle: Return empty list of events and the unchanged list of 'events_last_fetch_ids' for next run.
+
+    2.  All events from the current fetch cycle have the same timestamp.
+        Meaning: There are potentially more events with the same timestamp in the next fetch.
+        Handle: Add the list of fetched events IDs to current 'events_last_fetch_ids' from last run,
+                return list of new events and updated list of 'events_last_fetch_ids' for next run.
+
+    3.  Most recent event has later timestamp then other events in the response.
+        Meaning: This is the normal case where events in the response have different timestamps.
+        Handle: Return list of new events and a list of 'new_ids' containing only IDs of
+                events with identical latest time (up to second) for next run.
+
+    Args:
+        events (list[dict]): List of events from the current fetch response.
+        events_last_fetch_ids (list[str]): List of IDs of events from last fetch cycle.
+        unique_id_key (str): Unique event ID key of specific event type (Alert, Threat Activity etc.)
+
+    Returns:
+        tuple[list[dict], list[str]: The list of dedup events and ID list of events of current fetch.
+    """
+    # case 1
+    if not events:
+        demisto.debug('debug-log: Dedup case 1 - Empty event list (no new events received from API response).')
+        return [], events_last_fetch_ids
+
+    new_events: list[dict] = [event for event in events if event.get(unique_id_key) not in events_last_fetch_ids]
+
+    earliest_event_datetime = arg_to_datetime(events[0].get('time'))
+    latest_event_datetime = arg_to_datetime(events[-1].get('time'))
+
+    # case 2
+    if earliest_event_datetime and latest_event_datetime and\
+            are_two_datetime_equal_by_second(latest_event_datetime, earliest_event_datetime):
+        demisto.debug('debug-log: Dedup case 2 - All events from the current fetch cycle have the same timestamp.')
+        new_ids = [event.get(unique_id_key, '') for event in new_events]
+        events_last_fetch_ids.extend(new_ids)
+        return new_events, events_last_fetch_ids
+
+    # case 3
+    else:
+        demisto.debug('debug-log: Dedup case 3 - Most recent event has later timestamp then other events in the response.')
+        # Note that the following timestamps comparison are made between strings and assume
+        # the following timestamp format from the response: "YYYY-MM-DDTHH:MM:SS.000000+00:00"
```

Comment:
Its not. The API gives a full time up to milliseconds (not just zeros) even though we can only filter them up to seconds in the request. This was just a way to explain the time format, I see it can be misleading so I changed it to "YYYY-MM-DDTHH:MM:SS.fffff+Z".